buildingspy/development/optimica_run.template

```diff
@@ -29,174 +32,179 @@ def process_with_timeout(target, timeout):
-    mod = load_fmu(fmu_name)
-    x_nominal = mod.nominal_continuous_states
-
-    opts = mod.simulate_options() #Retrieve the default options
-    opts['logging'] = False
-    opts['solver'] = '{{ solver }}'
-    opts['ncp'] = {{ ncp }}
-
-    rtol = {{ rtol }}
 
-    if len(x_nominal) > 0:
-      atol = rtol*x_nominal
-    else:
-      atol = rtol
-
-    if opts['solver'] == 'CVode':
-      # Set user-specified tolerance if it is smaller than the tolerance in the .mo file
-      opts['CVode_options']['external_event_detection'] = False
-      opts['CVode_options']['maxh'] = (mod.get_default_experiment_stop_time()-mod.get_default_experiment_start_time())/float(opts['ncp'])
-      opts['CVode_options']['iter'] = 'Newton'
-      opts['CVode_options']['discr'] = 'BDF'
-      opts['CVode_options']['store_event_points'] = True # True is default, set to false if many events
-
-    if debug_solver:
-      opts["logging"] = True #<- Turn on solver debug logging
-      mod.set("_log_level", 4)
-
-
-    opts['{{ solver }}_options']['rtol'] = rtol
-    opts['{{ solver }}_options']['atol'] = atol
-    opts['filter'] = {{ filter }}
-    opts['result_file_name'] = "{{ result_file_name }}"
-
-
-    # Grab the std output
-    out = og.OutputGrabber()
-    out.start()
-    res = mod.simulate(
-        options=opts,
-        start_time={{ start_time }},
-        final_time={{ final_time }})
-    out.stop()
+def _translate(proc_num, return_dict):
+    import os
+    from pymodelica import compile_fmu
+    from inspect import signature
 
-    ######################################################################
-    # Get debugging information
-    if debug_solver and debug_solver_interactive_mode:
-      #Load the debug information
-      from pyfmi.debug import CVodeDebugInformation
-      debug = CVodeDebugInformation("{{ model }}".replace(".", "_")+"_debug.txt")
+    generate_html_diagnostics = {{generate_html_diagnostics}}
 
-      ### Below are options to plot the order, error and step-size evolution.
-      ### The error methods also take a threshold and a region if you want to
-      ### limit the plot to a certain interval.
+    try:
+        # OCT r28242 ignores MODELICAPATH and instead needs to have it set through
+        # a function argument.
+        compilation_log = "{{ model }}".replace('.', '_') + "_compile.log"
+        version = "2.0"
+        compiler_log_level = f"warning:{compilation_log}"
+        compiler_options = {"generate_html_diagnostics": generate_html_diagnostics,
+                            "nle_solver_tol_factor": 1e-2}  # 1e-2 is the default
+
+        sig = signature(compile_fmu)
+        if "modelicapath" in str(sig):
+            # This is the new API that uses modelicapath as a function argument.
+
+            removed_modelica_path = False
+            if 'MODELICAPATH' in os.environ:
+                modelicapath = os.environ['MODELICAPATH']
+                del os.environ['MODELICAPATH']
```

Comment:
@mwetter could it be the reason that in `OpenModelica` test that the `MODELICAPATH` is not set? Although I see that in file `BuildingsPy/buildingspy/tests/MyModelicaLibrary/Resources/Scripts/travis/bin/omc` it indeed tries to set the `MODELICAPATH`, and the `OpenModelica` tests are done before the `optimica` tests.